### PR TITLE
Add default build script, minor docker fixes

### PR DESCRIPTION
**Pull request by andy5995**
These commits should  not necessarily be squashed when merging. Your call.

The PR title could be changed though.

The changes here I think are fairly self-explanatory, with some details in the first commit. It'll probably make more sense if you first look at the two commits separately, rather than from the "files changed" tab.

### CODE REVIEW

File: scripts/build-default.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+# This file is part of vs-fltk
+# License: https://github.com/KaruroChori/vs-fltk/blob/master/LICENCE.md
+
+set -e
+
+if [ !-f "src/app/main.cpp" ]; then
```
Comment (KaruroChori) [inline scripts/build-default.sh:7]:
```suggestion
if [ !-f "src/app/main.cpp" ]; then
```
Sorry, just to check how it works

Comment (KaruroChori) [inline scripts/build-default.sh:7]:
```suggestion
if [ !-f "src/app/main.cpp" ]; then //Do not accept
```
Ok now I see.

**Comment by andy5995** [comment]
@KaruroChori This is ready for review.

**Comment by KaruroChori** [comment]
Hey, thanks! It also cleaned up the yml file quite a bit. I will have to add a note about that script in the dev docs as well, as it can simplify the annoying chain of commands.

**Comment by andy5995** [comment]
> Hey, thanks! It also cleaned up the yml file quite a bit. I will add a note about that script in the dev docs as well, as it can simplify the initial (annoying) chain of commands.

One problem with that script is that it takes longer because it reconfigures each time. I opened a related issue for that: https://github.com/KaruroChori/vs-fltk/issues/59

**Comment by KaruroChori** [comment]
There is a second problem I have been thinking. Meson subprojects should be probably updated?

**Comment by andy5995** [comment]
> There is a second problem I have been thinking. Meson subprojects should be probably updated at some point?

Hmmm... locally yes. That could be added to this build script for now, and see how we like it, change later if necessary. Could add a check so it doesn't bother to run if it's in the CI.

We could add a prompt to the script, so the user can just hit y or n to run it. Maybe a timeout.. I think that can be done with 'read'

ChatGPT says yes, and I think I've experimented with that before:

```sh
#!/bin/bash

read -t 4 -p "Do you want to proceed? (yes/no) " answer

# Check if input was received within 4 seconds
if [ $? -ne 0 ]; then
    echo "Timed out. Defaulting to 'no'."
    answer="no"
fi

# Normalize input (convert to lowercase) and handle response
case "${answer,,}" in
    yes)
        echo "You chose yes."
        ;;
    no)
        echo "You chose no."
        ;;
    *)
        echo "Invalid input. Defaulting to 'no'."
        ;;
esac
```

File: scripts/build-default.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  echo "This script must be run from the source root."
+  exit 1
+fi
+
+set -v
+
+bun install
+bun run codegen
+
+if [ -d "build" ]; then
```
Comment (andy5995) [inline scripts/build-default.sh:17]: Hmmm... this isn't working as expected when testing locally. Investigating...

Comment (andy5995) [reply scripts/build-default.sh:15]: Fixed.

File: scripts/build-default.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+set -e
+
+# Ensure the script is run from the source root
+if [ ! -f "src/app/main.cpp" ]; then
+  echo "This script must be run from the source root."
+  exit 1
+fi
+
+set -v
+if [ "CI" != "true" ]; then
```
Comment (andy5995) [inline scripts/build-default.sh:14]: I added this section to update subprojects as a separate commit. Let me know if you hate it and I'll revert it. ;) I liked this idea better than the prompt w/timeout.

**Comment by KaruroChori** [comment]
Thank you!

**Pull request merged**